### PR TITLE
Fix: Logic for Highlight/text-color format availability

### PR DESCRIPTION
**Pull request by dougwollison**
## What?
Corrects the logic for showing the text-color format, either if a palette is present or if custom colors are allowed, as expected.

## Why?
Previous logic mistakenly resulted in an unusable text color UI when palette was empty and custom was false, but offered no text color UI at all if palette was empty and custom was set to true.

## How?
Flipped test on allowCustomControl.

## Testing Instructions
1. Open theme.json
2. Under color settings, palette to [], custom to false
3. Open a post or page, insert a text block
4. Open the format dropdown, Highlight should be absent

For the inverse:
1. Set color.custom to true
2. Open editor, select/insert text block
3. Open format dropdown, Highlight should be present, selecting will allow custom color selection

### CODE REVIEW

**Comment by dougwollison** [comment]
Re-wrote after updating master on fork (swore I updated it beforehand). Conflict resolved.

File: packages/format-library/src/text-color/index.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ function TextColorEdit( {
 		[ contentRef, value, colors ]
 	);
 
-	const hasColorsToChoose = colors.length || ! allowCustomControl;
+	const hasColorsToChoose = colors.length || allowCustomControl;
```
Comment (Mamaduka) [inline packages/format-library/src/text-color/index.js]:
```suggestion
	const hasColorsToChoose = !! colors.length || allowCustomControl;
```

Suggestion (non-blocking): Let's cast the color length value to a Boolean. This ensures that `hasColorsToChoose` is always a boolean.

**Review by Mamaduka — APPROVED** [review]
Thanks for fixing this bug, @dougwollison!

File: packages/format-library/src/text-color/index.js (hunk 1 above)
Comment (dougwollison) [reply packages/format-library/src/text-color/index.js]: Done, seems it was originally boolean but when lodash was stripped out they left it as the bare `colors.length` without any tests.

**Pull request merged**